calblueprint/girls-write-now
# [save] Database schema changes for saving stories

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
* New table "Saved Stories" added to SupaBase to keep track of each user's saved stories per category "name"
* /src/queries/savedStories.tsx file created in with function that interact with the "Saved Stories" table
  * fetch functions created to fetch user saved stories
  * set functions created to set user saved stories

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links
N/A

### Notion Sprint Task
[https://www.notion.so/calblueprint/save-Database-schema-changes-for-saving-stories-4857b09d8e5f4388ab7b9854935b36fd](url)

[//]: # 'Add the relevant Notion link(s) here'

### Online sources
[https://supabase.com/docs/reference/javascript/update](url)
[https://supabase.com/dashboard/project/bkrdtgrnjmwbwqxfzjsr/editor/29743](url) (click < > API Docs)

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'

### Related PRs
N/A

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
* Check SupaBase table "Saved Stories"
* Go to /src/queries/savedStories.tsx and you can see the fetch and set functions created
  * There is one fetch function to fetch for user stories under the category "name" which is can be any category of user's choice
  * fetchUserStoriesFavorites and fetchUserStoriesReadingList are functions which fetch only the stories for the default categories "favorites" and "reading list"
  * There is one set function to set user stories under the category "name" which is can be any category of user's choice
  * setUserStoriesFavorites and setUserStoriesReadingList are functions which set only the stories for the default categories "favorites" and "reading list"
  * There is one delete function to delete any story per user associated with any category

[//]: # 'The order in which to review files and what to expect when testing locally'

## Next steps
* Create the front end component of this feature in order to view and add stories to the SupaBase per user
* Test that fetch, set, and delete methods work with front end component
* Add data to the SupaBase database using these methods

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
* Edge Case: To check if a user has no stories for the default categories "favorites" and "reading list", check the length of the outputted stories which were retrieved using any fetch method

[//]: # 'Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally'

### Screenshots
<img width="977" alt="Screenshot 2024-03-03 at 1 31 33 PM" src="https://github.com/calblueprint/girls-write-now/assets/69034384/f254ba8f-f9b4-4b7e-8024-90ce41dc28ef">
"Saved Stories" table in SupaBase

CC: @adityapawar1

[//]: # 'This tags in Adi as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
